swap_meet/vendor.py

```diff
@@ -1,2 +1,92 @@
+from types import NoneType
+from operator import attrgetter
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        #adds item to inventory
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        # removes item from inventory if it's already there
+        # if not, return false
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        # isolates all the items matching a specific category
+        # and returns them in the form of a list
+        self.category = category
+        category_list = []
+        for cat in self.inventory:
+            if cat.category == self.category:
+                category_list.append(cat)
+
+        return category_list
+
+    def get_best_by_category(self, category):
+        # determine which item is in the best condition
+        # for a given category
+        best_item = None
+        best_cat = 0
+
+        # utilizing the get_by_category function to 
+        # isolate the category
+        best_cat = self.get_by_category(category)
+        # if the function doesn't return a blank list
+        if best_cat:
+            # retreive the best condition
+
+            # the attrgetter allows me to isolate and evaluate the condition
+            # within the given instance
+            best_item = max(best_cat, key=attrgetter("condition"))
+        
+        return best_item
+
+    def swap_items(self, vendor2, my_item, their_item):
+        # takes an item from first list and puts in it second
+        # takes an item from second list and puts it in first
+        self.vendor2 = vendor2
+        self.my_item = my_item
+        self.their_item = their_item
+
+        if self.my_item not in self.inventory or self.their_item not in vendor2.inventory:
+            return False
+        else:
+            self.inventory.remove(my_item)
+            vendor2.inventory.append(my_item)
+            vendor2.inventory.remove(their_item)
+            self.inventory.append(their_item)
```

Comment:
you can use the instance methods you created to add and remove. So like `self.remove(my_item)`